### PR TITLE
[expo-sensors] Added Barometer

### PR DESCRIPTION
**Pull request by EvanBacon**
Moved Barometer from sensors-web PR

### CODE REVIEW

**Comment by EvanBacon** [comment]
This runs everywhere btw

File: docs/versions/unversioned/sdk/barometer.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
+---
+title: Barometer
+---
+
+Access the device barometer sensor to respond to changes in air pressure. Currently data is passed back as raw values. You will need to convert the units yourself. Below we've included information regarding the exact format data is returned in.
+
+| OS      | Units               | Provider                                                                                                | Description                                                                                                                         |
+| ------- | ------------------- | ------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------- |
+| iOS     | kilopascal          | [`CMAltimeter`](https://developer.apple.com/documentation/coremotion/cmaltimeter)                       | Altitude events reflect the change in the current altitude, not the absolute altitude.                                              |
```
Comment (fson) [inline docs/versions/unversioned/sdk/barometer.md]: Would it make sense to multiply the result by 10 on iOS to have the API return the pressure as hPa on both platforms? Having different units on different platforms seems annoying.

File: docs/versions/unversioned/sdk/barometer.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+#### Returns
+
+- A promise that resolves to a `boolean` denoting the availability of the sensor.
+
+### `Barometer.addListener(listener)`
+
+Subscribe for updates to the barometer.
+
+```js
+const subscription = Barometer.addListener(({ pressure /* iOS Only */, relativeAltitude }) => {
```
Comment (ide) [inline docs/versions/unversioned/sdk/barometer.md]: The way this is written, it looks like pressure is on iOS only. Maybe use newlines to make this less ambiguous:

```suggestion
const subscription = Barometer.addListener(({
  pressure,
  relativeAltitude, // iOS only
}) => {
```

File: docs/versions/unversioned/sdk/barometer.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+```
+
+#### Arguments
+
+- **listener (_function_)** -- A callback that is invoked when an barometer update is available. When invoked, the listener is provided a single argumument that is an object containing: `pressure: number`. On iOS the `relativeAltitude: number` value will also be available.
+
+#### Returns
+
+- A subscription that you can call `remove()` on when you
+  would like to unsubscribe the listener.
```
Comment (ide) [inline docs/versions/unversioned/sdk/barometer.md]: remove leading newline, let the browser wrap

File: docs/versions/unversioned/sdk/barometer.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    this._subscription = null;
+  };
+
+  render() {
+    const { pressure = 0 } = this.state.data;
+
+    return (
+      <View style={styles.sensor}>
+        <Text>Barometer:</Text>
+        <Text>{pressure * 100}</Text>
```
Comment (ide) [inline docs/versions/unversioned/sdk/barometer.md]:
```suggestion
        <Text>{pressure * 100} Pa</Text>
```

File: docs/versions/unversioned/sdk/barometer.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+```js
+const subscription = Barometer.addListener(({ pressure /* iOS Only */, relativeAltitude }) => {
+  console.log({ pressure, relativeAltitude });
+});
+```
+
+#### Arguments
+
+- **listener (_function_)** -- A callback that is invoked when an barometer update is available. When invoked, the listener is provided a single argumument that is an object containing: `pressure: number`. On iOS the `relativeAltitude: number` value will also be available.
```
Comment (ide) [inline docs/versions/unversioned/sdk/barometer.md]:
```suggestion
- **listener (_function_)** -- A callback that is invoked when an barometer update is available. When invoked, the listener is provided a single argument that is an object containing: `pressure: number`. On iOS the `relativeAltitude: number` value will also be available and is measured in meters.
```

File: packages/expo-sensors/ios/EXSensors/EXSensorsManager.m [hunk 6]
```diff
@@ -93,6 +105,45 @@ - (BOOL)isAccelerometerAvailable
...
+}
+
+- (void)setBarometerUpdateInterval:(NSTimeInterval)intervalMs
+{
+  // Do nothing
+}
+
+- (BOOL)isBarometerAvailable
+{
+  // TODO: Bacon: May need to check: ([CMPedometer authorizationStatus] == CMAuthorizationStatusAuthorized)
```
Comment (ide) [inline packages/expo-sensors/ios/EXSensors/EXSensorsManager.m]:
```suggestion
  // TODO: Bacon: May need to check: ([CMAltimeter authorizationStatus] == CMAuthorizationStatusAuthorized)
```

File: packages/expo-sensors/ios/EXSensors/Modules/SensorModules/EXBarometer.m [hunk 7]
```diff
@@ -0,0 +1,40 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+#import <EXSensors/EXBarometer.h>
+#import <EXSensorsInterface/EXBarometerInterface.h>
+
+@implementation EXBarometer
+
+EX_EXPORT_MODULE(ExponentBarometer);
```
Comment (ide) [inline packages/expo-sensors/ios/EXSensors/Modules/SensorModules/EXBarometer.m]:
```suggestion
EX_EXPORT_MODULE(ExpoBarometer);
```

File: packages/expo-sensors/src/Barometer.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import DeviceSensor from './DeviceSensor';
+import ExponentBarometer from './ExponentBarometer';
+
+type PressureMeasurement = {
```
Comment (ide) [inline packages/expo-sensors/src/Barometer.ts]:
```suggestion
type BarometerMeasurement = {
```

There are other types of pressure, like pressure on the touch screen and the pressure of unread GitHub notifications

**Review by ide — CHANGES_REQUESTED** [review]

File: docs/versions/unversioned/sdk/barometer.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+
+## Methods
+
+### `Barometer.isAvailableAsync()`
+
+Returns a promise which resolves into a boolean denoting the availability of the device barometer.
+
+| OS      | Availability                |
+| ------- | --------------------------- |
+| iOS     | iOS 8+                      |
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply docs/versions/unversioned/sdk/barometer.md]: Seems like it's nice to have some visibility into what this useless(?) method is doing

File: docs/versions/unversioned/sdk/barometer.md [hunk 10]
```diff
@@ -0,0 +1,155 @@
+---
+title: Barometer
+---
+
+Access the device barometer sensor to respond to changes in air pressure. `pressure` is converted to _`hectopascals`_ or _`hPa`_.
```
Comment (ide) [inline docs/versions/unversioned/sdk/barometer.md]:
```suggestion
Access the device barometer sensor to respond to changes in air pressure. `pressure` is measured in _`hectopascals`_ or _`hPa`_.
```

File: docs/versions/unversioned/sdk/barometer.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+
+> This module was introduced in Expo v33
+
+Barometer can be used outside of the Expo client with the `expo-sensors` & `expo-sensors-interface` libraries.
+
+## Methods
+
+### `Barometer.isAvailableAsync()`
+
+Returns a promise which resolves into a boolean denoting the availability of the device barometer.
```
Comment (ide) [inline docs/versions/unversioned/sdk/barometer.md]:
```suggestion
Returns a promise which resolves into a boolean denoting the availability of the device barometer. This does not account for whether the user has granted permission to get the barometer's measurements.
```

File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/modules/BarometerModule.java [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+  @Override
+  protected SensorService getSensorService() {
+    return getModuleRegistry().getModule(BarometerService.class);
+  }
+
+  protected Bundle eventToMap(SensorEvent sensorEvent) {
+    Bundle map = new Bundle();
+    // TODO: Bacon: Can we get relative altitude?
+    // TODO: Bacon: Should we create a conversion from PSI to hPa?
```
Comment (ide) [inline packages/expo-sensors/android/src/main/java/expo/modules/sensors/modules/BarometerModule.java]: Is this PSI comment still relevant?

**Review by ide — APPROVED** [review]

File: docs/versions/unversioned/sdk/barometer.md (hunk 11 above)
Comment (EvanBacon) [reply docs/versions/unversioned/sdk/barometer.md]: I still cannot find anything regarding permission. It doesn't look like you need anything in the `info.plist`. I suppose that would mean this function accounts for all availability cases.

**Pull request merged**